### PR TITLE
Fix rules

### PR DESCRIPTION
**Pull request by rambleraptor**
This fixes all of the rules except the parent rule.

### CODE REVIEW

File: rules/aep0004/resource_plural.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ var resourcePlural = &lint.MessageRule{
 		r := utils.GetResource(m)
 		l := locations.MessageResource(m)
 		p := r.GetPlural()
-		pLower := utils.ToLowerCamelCase(p)
+		pLower := utils.ToKebobCase(p)
```
Comment (rofrankel) [inline rules/aep0004/resource_plural.go:34]:
```suggestion
		pLower := utils.ToKebabCase(p)
```

Here and elsewhere :)

**Review by rofrankel — CHANGES_REQUESTED** [review]

File: rules/aep0004/resource_name_components_alternate.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/jhump/protoreflect/desc"
 )
 
-var identifierRegexp = regexp.MustCompile("^{[a-z][_a-z0-9]*[a-z0-9]}$")
+var identifierRegexp = regexp.MustCompile("^{[a-z][-_a-z0-9]*[a-z0-9]}$")
```
Comment (toumorokoshi) [inline rules/aep0004/resource_name_components_alternate.go:28]:
```suggestion
var identifierRegexp = regexp.MustCompile("^{[a-z][-a-z0-9]*[a-z0-9]}$")
```
I think we shouldn't allow underscores then? it must be kebab-case.

File: rules/aep0004/resource_singular.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ var resourceSingular = &lint.MessageRule{
 		l := locations.MessageResource(m)
 		s := r.GetSingular()
 		_, typeName, ok := utils.SplitResourceTypeName(r.GetType())
-		lowerTypeName := utils.ToLowerCamelCase(typeName)
+		lowerTypeName := utils.ToKebobCase(typeName)
```
Comment (toumorokoshi) [inline rules/aep0004/resource_singular.go:35]:
```suggestion
		lowerTypeName := utils.ToKebabCase(typeName)
```
https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case - small typo I think.

File: rules/aep0004/resource_type_name.go [hunk 4]
```diff
@@ -32,17 +32,17 @@ var resourceTypeName = &lint.MessageRule{
 	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
 		resource := utils.GetResource(m)
 		_, typeName, ok := utils.SplitResourceTypeName(resource.GetType())
-		upperTypeName := utils.ToUpperCamelCase(typeName)
+		kebob_case := utils.ToKebobCase(typeName)
```
Comment (toumorokoshi) [inline rules/aep0004/resource_type_name.go:35]:
```suggestion
		kebabCase := utils.ToKebobCase(typeName)
```
go variable naming is camelCase

File: rules/internal/utils/casing.go [hunk 5]
```diff
@@ -14,54 +14,27 @@
...
-
-// ToLowerCamelCase returns the lowerCamelCase of a string, including removing
-// delimiters (_,-,., ) and using them to denote a new word.
-func ToLowerCamelCase(s string) string {
-	return toCamelCase(s, false, true)
-}
-
-func toCamelCase(s string, makeNextUpper bool, makeNextLower bool) string {
+// ToKebobCase returns the kebob-case of a word (book-edition).
+func ToKebobCase(s string) string {
```
Comment (toumorokoshi) [inline rules/internal/utils/casing.go:18]: Not a big deal, but would be good to upstream this and other casing to aep-lib-go: https://github.com/aep-dev/aep-lib-go/blob/main/pkg/cases/cases.go

**Review by toumorokoshi — APPROVED** [review]
a few nits I'd like fixed, but approving assuming you'll address those. Thanks!

File: rules/internal/utils/casing.go (hunk 5 above)
Comment (rambleraptor) [reply rules/internal/utils/casing.go:18]: I can do that in a future PR.

**Pull request merged**